pycqed/analysis_v2/coherence_analysis.py

```diff
@@ -1,27 +1,302 @@
 '''
 Hacked together by Rene Vollmer
+Cleaned up (a bit) by Adriaan
 '''
-
+import os
+import matplotlib.pyplot as plt
 import datetime
+from collections import OrderedDict
+from copy import deepcopy
 import pycqed.analysis_v2.base_analysis as ba
-from pycqed.analysis_v2.base_analysis import plot_scatter_errorbar_fit, plot_scatter_errorbar
-from pycqed.analysis import measurement_analysis as ma_old
+from pycqed.analysis_v2.base_analysis import plot_scatter_errorbar_fit,\
+    plot_scatter_errorbar, set_xlabel, set_ylabel
 
 import numpy as np
 import lmfit
+from pycqed.analysis.tools.plotting import SI_val_to_msg_str, \
+    format_lmfit_par, plot_lmfit_res
 
 from pycqed.analysis import analysis_toolbox as a_tools
 
 
+class CoherenceAnalysis(ba.BaseDataAnalysis):
+    """
+    Power spectral density analysis of transmon coherence.
+
+    Note, analysis of the coherence times is separated from data extraction
+    as that can be a complicated process that is highly experiment dependent.
+
+    Args:
+        coherence_table : table containing the data, see below for
+            specification.
+        freq_resonator: readout resonator frequency (in Hz)
+        Qc:             coupling Q of the readout resonator
+        chi_shift       in Hz
+        savename:       figures are saved in a folder created based on the
+            provided t_stop in the datadir. `savename` is used to name
+            this folder.
+
+
+    Coherence_table specification:
+           Row  | Content (arrays)
+        --------+--------
+            1   | dac
+            2   | frequency (Hz)
+            3   | T1 (s)
+            4   | T2 star (s)
+            5   | T2 echo (s)
+            6   | Exclusion mask (True where data is to be excluded)
+
+    Generates 8 plots:
+        > T1, T2, Echo vs flux
+        > T1, T2, Echo vs frequency
+        > T1, T2, Echo vs flux sensitivity
+        > ratio Ramsey/Echo vs flux
+        > ratio Ramsey/Echo vs frequency
+        > ratio Ramsey/Echo vs flux sensitivity
+        > Dephasing rates Ramsey and Echo vs flux sensitivity
+        > Dac arc fit, use to assess if sensitivity calculation is correct.
```

Comment:
Add description of where in the analysis object the relevant parameters are stored. Half of the time when I use analysis object I don't care about the plots, only about fitted parameters. Say I want to use \sqrt(A) for ramsey. I'd rather not dig through the code to figure out how to get access to that in my analysis notebook.

Reply:
(Philosophical comment) This is one of the reasons why everyone writes their own scripts. It is less frustrating and easier to write your own script then to figure out where in the analysis object the results are stored.

Reply:
Excellent point.